bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/api/dto/clip2/helper/Setters.java

```diff
@@ -341,7 +349,67 @@ public static Resource setResource(Resource target, Resource source) {
                 targetTimedEffects.setDuration(duration);
             }
         }
-
         return target;
     }
+
+    /**
+     * Merge on/dimming/color fields from light and grouped light resources.
+     *
+     * Subsequent sparse resources will be merged into the first one.
+     * Sparse resources will be merged into any prior full state resources.
+     * Full state resources will replace any prior sparse resources.
+     */
+    public static Collection<Resource> mergeLightResources(Collection<Resource> resources) {
+        Map<String, Resource> resourceIndex = new HashMap<>();
+        Iterator<Resource> iterator = resources.iterator();
+        while (iterator.hasNext()) {
+            Resource resource = iterator.next();
+            String id = resource.getId();
+            Resource indexedResource = resourceIndex.get(id);
+            if (indexedResource == null) {
+                indexedResource = resource;
+                resourceIndex.put(id, indexedResource);
+            }
+
+            if (!LIGHT_TYPES.contains(resource.getType()) || !resource.hasHSBComponent()
+                    || resource.equals(indexedResource)) {
+                continue;
+            }
+
+            // Full state takes precedence, invalidate previous sparse resources.
+            if (resource.hasFullState()) {
+                indexedResource.setOnState(null);
+                indexedResource.setDimming(null);
+                indexedResource.setColorXy(null);
+
+                resourceIndex.put(id, resource);
+                continue;
+            }
+
+            OnState onState = resource.getOnState();
+            if (onState != null) {
+                indexedResource.setOnState(onState);
+                resource.setOnState(null);
+            }
+            Dimming dimming = resource.getDimming();
+            if (dimming != null) {
+                indexedResource.setDimming(dimming);
+                resource.setDimming(null);
+            }
+            ColorXy colorXy = resource.getColorXy();
+            if (colorXy != null) {
+                indexedResource.setColorXy(colorXy);
+                resource.setColorXy(null);
+            }
+
```

Comment:
I think you should probably do the same as above (for OnOff, Dimming, ColorXy fields) for the ColorTemperature field. Admittedly it is not directly HSB related, but if the light has been commanded from a colorXy co-ordinate that is off the 'white color locus' to one that is on the locus (or vice versa) then the SSE DTO will contain both a ColorXy field and also a ColorTemperature field. And if one de-/re-links the former field, then one should also de-/re-link the latter.

Note: this is slightly linked to my comment below about merging / not merging of 'complex' resources.

Reply:
This is related to the discussion above, so to not duplicate I'll resolve this one.